src/python/pants/option/global_options.py

```diff
@@ -874,11 +874,14 @@ def register_options(cls, register):
             "tags ('-' prefix).  Useful with ::, to find subsets of targets "
             "(e.g., integration tests.)",
         )
+        ci_var = "CI" in os.environ
         register(
             "--dynamic-ui",
             type=bool,
-            default=sys.stderr.isatty(),
-            help="Display a dynamically-updating console UI as pants runs.",
+            default=((not ci_var) and sys.stderr.isatty()),
+            help="Display a dynamically-updating console UI as pants runs. This is true by default "
+            "if pants detects a TTY and there is no 'CI' environment variable indicating that "
```

Comment:
Nit: best to capitalize Pants because this shows up in documentation for end users.